### PR TITLE
Update Earl Grey meta.yml

### PR DESCRIPTION
**Pull request by TobyBaril**
Describe your pull request here

Add EMBOSS and pybedtools as requirements for the environment. 
----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by TobyBaril** [comment]
@bioconda/core test failing for out of space on test pipeline (common problem with RepeatMasker in Conda). Please advise on how to push this change as these updated requirements are needed for Earl Grey

`{"message":"write /usr/local/share/RepeatMasker/Libraries/RepeatAnnotationData.pm: no space left on device"}`

File: recipes/earlgrey/meta.yaml [hunk 1]
```diff
@@ -11,7 +11,7 @@ source:
   sha256: {{ sha256 }}
 
 build:
-  number: 0
+  number: 5
```
Comment (bgruening) [inline recipes/earlgrey/meta.yaml:14]:
```suggestion
  number: 1
```

**Comment by TobyBaril** [comment]
Lots of errors in the linux test, related to permissions and some other bits I think? I'm not sure that this is linked to any changes made here though

```
13:55:00 BIOCONDA ERROR COMMAND FAILED (exited with 1): mulled-build build-and-test earlgrey=4.1.1--h4ac6f70_1 -n biocontainers --test bash -c '/usr/local/env-execute true && . /usr/local/env-activate.sh && earlGrey -h' --channels conda-forge,file:///opt/mambaforge/envs/bioconda/conda-bld,bioconda,defaults --involucro-path /opt/mambaforge/envs/bioconda/lib/python3.9/site-packages/bioconda_utils/involucro

Traceback (most recent call last):
  File "/opt/mambaforge/envs/bioconda/lib/python3.9/site-packages/bioconda_utils/pkg_test.py", line 180, in test_package
    p = utils.run(cmd, env=env, cwd=d, mask=False, live=live_logs)
  File "/opt/mambaforge/envs/bioconda/lib/python3.9/site-packages/bioconda_utils/utils.py", line 697, in run
    raise sp.CalledProcessError(returncode, masked_cmds, output=output)
subprocess.CalledProcessError: Command '['mulled-build', 'build-and-test', 'earlgrey=4.1.1--h4ac6f70_1', '-n', 'biocontainers', '--test', "bash -c '/usr/local/env-execute true && . /usr/local/env-activate.sh && earlGrey -h'", '--channels', 'conda-forge,file:///opt/mambaforge/envs/bioconda/conda-bld,bioconda,defaults', '--involucro-path', '/opt/mambaforge/envs/bioconda/lib/python3.9/site-packages/bioconda_utils/involucro']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/mambaforge/envs/bioconda/lib/python3.9/shutil.py", line 688, in _rmtree_safe_fd
    os.unlink(entry.name, dir_fd=topfd)
PermissionError: [Errno 13] Permission denied: 'nm'
```

**Comment by TobyBaril** [comment]
Build is successful, but same space errors in log as @wjunLu, any way to resolve and merge this change?

Screenshot 2024-04-26 at 16 03 53

Screenshot 2024-04-26 at 16 04 27

**Comment by TobyBaril** [comment]
@mencian @bgruening @bioconda/core , I'm having a lot of trouble with the Azure pipeline running out of space after a successful build, as are a few others like @wjunLu. This has been an issue with more complex packages for a while, is there any solution to get this update pushed - it is fundamental to the next release of the pipeline?

```
2024-04-29T07:33:49.3500091Z 07:33:47 [32mBIOCONDA INFO[0m BUILD SUCCESS earlgrey-4.1.1-h4ac6f70_1.tar.bz2[0m
2024-04-29T07:33:49.3500500Z 07:33:47 [32mBIOCONDA INFO[0m TEST START via mulled-build recipes/earlgrey[0m
2024-04-29T07:34:49.3435490Z 07:33:49 [32mBIOCONDA INFO[0m (COMMAND) mulled-build build-and-test earlgrey=4.1.1--h4ac6f70_1 -n biocontainers --test bash -c '/usr/local/env-execute true && . /usr/local/env-activate.sh && earlGrey -h' --channels conda-forge,file:///opt/mambaforge/envs/bioconda/conda-bld,bioconda,defaults --involucro-path /opt/mambaforge/envs/bioconda/lib/python3.9/site-packages/bioconda_utils/involucro[0m
2024-04-29T07:34:49.3437449Z 07:33:50 [32mBIOCONDA INFO[0m (ERR) Passing program arguments as a string may be a security hazard if combined with untrusted input[0m
2024-04-29T07:34:49.3440872Z 07:33:50 [32mBIOCONDA INFO[0m (OUT) /opt/mambaforge/envs/bioconda/lib/python3.9/site-packages/bioconda_utils/involucro -v=2 -f /opt/mambaforge/envs/bioconda/lib/python3.9/site-packages/galaxy/tools/deps/mulled/invfile.lua -set CHANNELS='conda-forge,file:///opt/mambaforge/envs/bioconda/conda-bld,bioconda,defaults' -set TARGETS='earlgrey=4.1.1=h4ac6f70_1' -set REPO='quay.io/biocontainers/earlgrey:4.1.1--h4ac6f70_1' -set BINDS='build/dist:/usr/local/,/opt/mambaforge/envs/bioconda/conda-bld:/opt/mambaforge/envs/bioconda/conda-bld' -set DEST_BASE_IMAGE='quay.io/bioconda/base-glibc-busybox-bash:3.0' -set CONDA_IMAGE='quay.io/bioconda/create-env:latest' -set TEST='bash -c '"'"'/usr/local/env-execute true && . /usr/local/env-activate.sh && earlGrey -h'"'"'' build-and-test[0m
2024-04-29T07:34:49.3443865Z 07:33:50 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:33:50] DEBU Run file [/opt/mambaforge/envs/bioconda/lib/python3.9/site-packages/galaxy/tools/deps/mulled/invfile.lua][0m
2024-04-29T07:34:49.3444904Z 07:33:50 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:33:50] INFO Invoke Task [build][0m
2024-04-29T07:34:49.3446042Z 07:33:50 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:33:50] STEP Run image [quay.io/bioconda/create-env:latest] with command [[rm -rf /data/dist]][0m
2024-04-29T07:34:49.3447113Z 07:33:50 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:33:50] DEBU Creating container [step-df06eaafb0][0m
2024-04-29T07:34:49.3448206Z 07:33:50 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:33:50] DEBU Image [quay.io/bioconda/create-env:latest] not present, pulling it[0m
2024-04-29T07:34:49.3449293Z 07:33:50 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:33:50] DEBU Pull Image [quay.io/bioconda/create-env:latest][0m
2024-04-29T07:34:49.3450377Z 07:33:58 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:33:58] DEBU Created container [c08e26074e05 step-df06eaafb0], starting it[0m
2024-04-29T07:34:49.3451534Z 07:33:59 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:33:59] DEBU Container [c08e26074e05 step-df06eaafb0] started, waiting for completion[0m
2024-04-29T07:34:49.3452780Z 07:33:59 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:33:59] DEBU Container [c08e26074e05 step-df06eaafb0] completed with exit code [0] as expected[0m
2024-04-29T07:34:49.3454161Z 07:33:59 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:33:59] DEBU Container [c08e26074e05 step-df06eaafb0] removed[0m
2024-04-29T07:34:49.3456055Z 07:33:59 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:33:59] STEP Run image [quay.io/bioconda/create-env:latest] with command [[/bin/sh -c conda() { mamba "${@}" ; } && conda install  -c conda-forge -c file:///opt/mambaforge/envs/bioconda/conda-bld -c bioconda -c defaults  earlgrey=4.1.1=h4ac6f70_1 -p /usr/local --copy --yes --quiet&&create-env --conda=: /usr/local]][0m
2024-04-29T07:34:49.3457435Z 07:33:59 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:33:59] DEBU Creating container [step-9a814d91c6][0m
2024-04-29T07:34:49.3458932Z 07:33:59 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:33:59] DEBU Created container [19acaa0fea72 step-9a814d91c6], starting it[0m
2024-04-29T07:34:49.3479880Z 07:33:59 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:33:59] DEBU Container [19acaa0fea72 step-9a814d91c6] started, waiting for completion[0m
2024-04-29T07:38:49.3438138Z ....07:38:40 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:38:40] SOUT Preparing transaction: ...working... done[0m
2024-04-29T07:39:41.2864026Z ##[warning]Free disk space on / is lower than 5%; Currently used: 95.06%
2024-04-29T07:39:42.0918667Z ##[warning]Free disk space on / is lower than 5%; Currently used: 95.20%
2024-04-29T07:39:42.0959756Z ##[warning]Free disk space on / is lower than 5%; Currently used: 95.20%
2024-04-29T07:39:42.2287909Z ##[warning]Free disk space on / is lower than 5%; Currently used: 95.20%
2024-04-29T07:39:43.0365182Z ##[warning]Free disk space on / is lower than 5%; Currently used: 95.34%
2024-04-29T07:39:43.0369058Z ##[warning]Free disk space on / is lower than 5%; Currently used: 95.34%
2024-04-29T07:39:44.0702884Z ##[warning]Free disk space on / is lower than 5%; Currently used: 95.51%
2024-04-29T07:39:49.3439390Z .07:39:04 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:39:04] SOUT Verifying transaction: ...working... done[0m
2024-04-29T07:40:49.3441552Z .07:40:17 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:40:17] SOUT Executing transaction: ...working... To install the databases needed by RepeatMasker, execute:[0m
2024-04-29T07:40:49.3442405Z 07:40:17 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:40:17] SOUT[0m
2024-04-29T07:40:49.3443443Z 07:40:17 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:40:17] SOUT wget -O Dfam_curatedonly.h5.gz https://www.dfam.org/releases/Dfam_3.7/families/Dfam_curatedonly.h5.gz[0m
2024-04-29T07:40:49.3444196Z 07:40:17 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:40:17] SOUT gunzip -c Dfam_curatedonly.h5.gz > /usr/local/share/RepeatMasker/Libraries/Dfam.h5[0m
2024-04-29T07:40:49.3444802Z 07:40:17 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:40:17] SOUT rm Dfam_curatedonly.h5.gz[0m
2024-04-29T07:40:49.3445281Z 07:40:17 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:40:17] SOUT[0m
2024-04-29T07:40:49.3445754Z 07:40:17 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:40:17] SOUT[0m
2024-04-29T07:40:49.3446330Z 07:40:17 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:40:17] SOUT done[0m
2024-04-29T07:40:49.3446876Z 07:40:17 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:40:17] SERR creating environment at /usr/local ...[0m
2024-04-29T07:40:49.3447444Z 07:40:17 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:40:17] SERR generating activation script...[0m
2024-04-29T07:40:49.3448058Z 07:40:18 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:40:18] SERR writing activation script to /usr/local/env-activate.sh ...[0m
2024-04-29T07:40:49.3448688Z 07:40:18 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:40:18] SERR writing execution script to /usr/local/env-execute ...[0m
2024-04-29T07:40:49.3449281Z 07:40:18 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:40:18] SERR copying license files to /usr/local/conda-meta ...[0m
2024-04-29T07:40:49.3449852Z 07:40:21 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:40:21] SERR finished create-env for /usr/local[0m
2024-04-29T07:40:49.3450490Z 07:40:23 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:40:23] DEBU Container [19acaa0fea72 step-9a814d91c6] completed with exit code [0] as expected[0m
2024-04-29T07:40:49.3451286Z 07:40:25 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:40:25] DEBU Container [19acaa0fea72 step-9a814d91c6] removed[0m
2024-04-29T07:40:49.3451933Z 07:40:25 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:40:25] STEP Wrap [build/dist] as [quay.io/biocontainers/earlgrey:4.1.1--h4ac6f70_1][0m
2024-04-29T07:40:49.3452549Z 07:40:25 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:40:25] DEBU Creating container [step-36e00c9750][0m
2024-04-29T07:40:49.3453180Z 07:40:25 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:40:25] DEBU Image [quay.io/bioconda/base-glibc-busybox-bash:3.0] not present, pulling it[0m
2024-04-29T07:40:49.3453844Z 07:40:25 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:40:25] DEBU Pull Image [quay.io/bioconda/base-glibc-busybox-bash:3.0][0m
2024-04-29T07:40:49.3454600Z 07:40:39 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:40:39] DEBU Packing failed due to open build/dist/conda-meta/font-ttf-ubuntu-0.83-h77eed37_1/licenses/LICENCE.txt: permission denied[0m
2024-04-29T07:40:49.3455494Z 07:40:40 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:40:40] WARN Local execution errorred with error [open build/dist/conda-meta/font-ttf-ubuntu-0.83-h77eed37_1/licenses/LICENCE.txt: permission denied], retrying with remote execution[0m
2024-04-29T07:40:49.3456208Z 07:40:40 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:40:40] DEBU Creating container [step-8b5fac5242][0m
2024-04-29T07:40:49.3456808Z 07:40:40 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:40:40] DEBU Image [involucro/tool:latest] not present, pulling it[0m
2024-04-29T07:40:49.3457375Z 07:40:40 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:40:40] DEBU Pull Image [involucro/tool:latest][0m
2024-04-29T07:40:49.3457996Z 07:40:42 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:40:42] DEBU Created container [2c90a59348e1 step-8b5fac5242], starting it[0m
2024-04-29T07:40:49.3458649Z 07:40:42 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:40:42] DEBU Container [2c90a59348e1 step-8b5fac5242] started, waiting for completion[0m
2024-04-29T07:40:49.3459312Z 07:40:42 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:40:42] SERR [Apr 29 07:40:42] DEBU Creating container [step-c0ff258d46][0m
2024-04-29T07:41:49.3442980Z .07:41:34 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:41:34] SERR [Apr 29 07:41:34] DEBU Packing succeeded[0m
2024-04-29T07:42:49.3447786Z .07:42:38 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:42:38] SERR [Apr 29 07:42:38] ERRO Task processing failed: API error (500): {"message":"write /usr/local/share/LTR_retriever/database/alluniRefprexp082813: no space left on device"}[0m
2024-04-29T07:42:49.3449589Z 07:42:38 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:42:38] SERR[0m
2024-04-29T07:42:49.3451646Z 07:42:39 [32mBIOCONDA INFO[0m (ERR) [Apr 29 07:42:39] ERRO Task processing failed: Unexpected exit code [1] of container [2c90a59348e1 step-8b5fac5242], container preserved[0m
2024-04-29T07:42:51.4586668Z .07:42:51 [32mBIOCONDA INFO[0m (OUT) .........[0m
2024-04-29T07:42:51.4587611Z 07:42:51 [32mBIOCONDA INFO[0m (ERR) Traceback (most recent call last):[0m
2024-04-29T07:42:51.4588223Z 07:42:51 [32mBIOCONDA INFO[0m (ERR)   File "/opt/mambaforge/envs/bioconda/bin/mulled-build", line 10, in <module>[0m
2024-04-29T07:42:51.4588692Z 07:42:51 [32mBIOCONDA INFO[0m (ERR)     sys.exit(main())[0m
2024-04-29T07:42:51.4589453Z 07:42:51 [32mBIOCONDA INFO[0m (ERR)   File "/opt/mambaforge/envs/bioconda/lib/python3.9/site-packages/galaxy/tools/deps/mulled/mulled_build.py", line 414, in main[0m
2024-04-29T07:42:51.4590074Z 07:42:51 [32mBIOCONDA INFO[0m (ERR)     sys.exit(mull_targets(targets, **args_to_mull_targets_kwds(args)))[0m
2024-04-29T07:42:51.4590766Z 07:42:51 [32mBIOCONDA INFO[0m (ERR)   File "/opt/mambaforge/envs/bioconda/lib/python3.9/site-packages/galaxy/tools/deps/mulled/mulled_build.py", line 243, in mull_targets[0m
2024-04-29T07:42:51.4591361Z 07:42:51 [32mBIOCONDA INFO[0m (ERR)     ret = involucro_context.exec_command(involucro_args)[0m
2024-04-29T07:42:51.4592101Z 07:42:51 [32mBIOCONDA INFO[0m (ERR)   File "/opt/mambaforge/envs/bioconda/lib/python3.9/site-packages/galaxy/tools/deps/mulled/mulled_build.py", line 287, in exec_command[0m
2024-04-29T07:42:51.4593112Z 07:42:51 [32mBIOCONDA INFO[0m (ERR)     shutil.rmtree('./build')[0m
2024-04-29T07:42:51.4593648Z 07:42:51 [32mBIOCONDA INFO[0m (ERR)   File "/opt/mambaforge/envs/bioconda/lib/python3.9/shutil.py", line 734, in rmtree[0m
2024-04-29T07:42:51.4594172Z 07:42:51 [32mBIOCONDA INFO[0m (ERR)     _rmtree_safe_fd(fd, path, onerror)[0m
2024-04-29T07:42:51.4594732Z 07:42:51 [32mBIOCONDA INFO[0m (ERR)   File "/opt/mambaforge/envs/bioconda/lib/python3.9/shutil.py", line 667, in _rmtree_safe_fd[0m
2024-04-29T07:42:51.4595452Z 07:42:51 [32mBIOCONDA INFO[0m (ERR)     _rmtree_safe_fd(dirfd, fullname, onerror)[0m
2024-04-29T07:42:51.4596109Z 07:42:51 [32mBIOCONDA INFO[0m (ERR)   File "/opt/mambaforge/envs/bioconda/lib/python3.9/shutil.py", line 667, in _rmtree_safe_fd[0m
2024-04-29T07:42:51.4596669Z 07:42:51 [32mBIOCONDA INFO[0m (ERR)     _rmtree_safe_fd(dirfd, fullname, onerror)[0m
2024-04-29T07:42:51.4597269Z 07:42:51 [32mBIOCONDA INFO[0m (ERR)   File "/opt/mambaforge/envs/bioconda/lib/python3.9/shutil.py", line 667, in _rmtree_safe_fd[0m
2024-04-29T07:42:51.4597785Z 07:42:51 [32mBIOCONDA INFO[0m (ERR)     _rmtree_safe_fd(dirfd, fullname, onerror)[0m
2024-04-29T07:42:51.4598371Z 07:42:51 [32mBIOCONDA INFO[0m (ERR)   File "/opt/mambaforge/envs/bioconda/lib/python3.9/shutil.py", line 690, in _rmtree_safe_fd[0m
2024-04-29T07:42:51.4598892Z 07:42:51 [32mBIOCONDA INFO[0m (ERR)     onerror(os.unlink, fullname, sys.exc_info())[0m
2024-04-29T07:42:51.4599476Z 07:42:51 [32mBIOCONDA INFO[0m (ERR)   File "/opt/mambaforge/envs/bioconda/lib/python3.9/shutil.py", line 688, in _rmtree_safe_fd[0m
2024-04-29T07:42:51.4599977Z 07:42:51 [32mBIOCONDA INFO[0m (ERR)     os.unlink(entry.name, dir_fd=topfd)[0m
2024-04-29T07:42:51.4600478Z 07:42:51 [32mBIOCONDA INFO[0m (ERR) PermissionError: [Errno 13] Permission denied: 'nm'[0m
2024-04-29T07:42:51.4600976Z 07:42:51 [32mBIOCONDA INFO[0m Command closed STDOUT/STDERR but is still running[0m
2024-04-29T07:42:51.4601408Z 07:42:51 [32mBIOCONDA INFO[0m Waiting 30 seconds (1/5)[0m
2024-04-29T07:42:51.4602509Z 07:42:51 [31mBIOCONDA ERROR[0m COMMAND FAILED (exited with 1): mulled-build build-and-test earlgrey=4.1.1--h4ac6f70_1 -n biocontainers --test bash -c '/usr/local/env-execute true && . /usr/local/env-activate.sh && earlGrey -h' --channels conda-forge,file:///opt/mambaforge/envs/bioconda/conda-bld,bioconda,defaults --involucro-path /opt/mambaforge/envs/bioconda/lib/python3.9/site-packages/bioconda_utils/involucro[0m
2024-04-29T07:42:51.4603250Z 
2024-04-29T07:42:51.4640062Z Traceback (most recent call last):
2024-04-29T07:42:51.4640656Z   File "/opt/mambaforge/envs/bioconda/lib/python3.9/site-packages/bioconda_utils/pkg_test.py", line 180, in test_package
2024-04-29T07:42:51.4640963Z     p = utils.run(cmd, env=env, cwd=d, mask=False, live=live_logs)
2024-04-29T07:42:51.4641376Z   File "/opt/mambaforge/envs/bioconda/lib/python3.9/site-packages/bioconda_utils/utils.py", line 697, in run
2024-04-29T07:42:51.4641685Z     raise sp.CalledProcessError(returncode, masked_cmds, output=output)
2024-04-29T07:42:51.4643017Z subprocess.CalledProcessError: Command '['mulled-build', 'build-and-test', 'earlgrey=4.1.1--h4ac6f70_1', '-n', 'biocontainers', '--test', "bash -c '/usr/local/env-execute true && . /usr/local/env-activate.sh && earlGrey -h'", '--channels', 'conda-forge,file:///opt/mambaforge/envs/bioconda/conda-bld,bioconda,defaults', '--involucro-path', '/opt/mambaforge/envs/bioconda/lib/python3.9/site-packages/bioconda_utils/involucro']' returned non-zero exit status 1.
2024-04-29T07:42:51.4643527Z 
2024-04-29T07:42:51.4643712Z During handling of the above exception, another exception occurred:
2024-04-29T07:42:51.4643821Z 
2024-04-29T07:42:51.4643982Z Traceback (most recent call last):
2024-04-29T07:42:51.4644209Z   File "/opt/mambaforge/envs/bioconda/lib/python3.9/shutil.py", line 688, in _rmtree_safe_fd
2024-04-29T07:42:51.4644670Z     os.unlink(entry.name, dir_fd=topfd)
2024-04-29T07:42:51.4644867Z PermissionError: [Errno 13] Permission denied: 'nm'
2024-04-29T07:42:51.4644962Z 
2024-04-29T07:42:51.4645154Z During handling of the above exception, another exception occurred:
2024-04-29T07:42:51.4645264Z 
2024-04-29T07:42:51.4645407Z Traceback (most recent call last):
2024-04-29T07:42:51.4645744Z   File "/opt/mambaforge/envs/bioconda/bin/bioconda-utils", line 8, in <module>
2024-04-29T07:42:51.4659734Z     sys.exit(main())
2024-04-29T07:42:51.4660404Z   File "/opt/mambaforge/envs/bioconda/lib/python3.9/site-packages/bioconda_utils/cli.py", line 1121, in main
2024-04-29T07:42:51.4660745Z     argh.dispatch_commands([
2024-04-29T07:42:51.4661192Z   File "/opt/mambaforge/envs/bioconda/lib/python3.9/site-packages/argh/dispatching.py", line 328, in dispatch_commands
2024-04-29T07:42:51.4679959Z     dispatch(parser, *args, **kwargs)
2024-04-29T07:42:51.4680494Z   File "/opt/mambaforge/envs/bioconda/lib/python3.9/site-packages/argh/dispatching.py", line 174, in dispatch
2024-04-29T07:42:51.4680738Z     for line in lines:
2024-04-29T07:42:51.4685414Z   File "/opt/mambaforge/envs/bioconda/lib/python3.9/site-packages/argh/dispatching.py", line 277, in _execute_command
2024-04-29T07:42:51.4685685Z     for line in result:
2024-04-29T07:42:51.4686083Z   File "/opt/mambaforge/envs/bioconda/lib/python3.9/site-packages/argh/dispatching.py", line 260, in _call
2024-04-29T07:42:51.4698307Z     result = function(*positional, **keywords)
2024-04-29T07:42:51.4703279Z   File "<boltons.funcutils.FunctionBuilder-5>", line 2, in build
2024-04-29T07:42:51.4714440Z   File "/opt/mambaforge/envs/bioconda/lib/python3.9/site-packages/bioconda_utils/cli.py", line 137, in wrapper
2024-04-29T07:42:51.4714753Z     func(*args, **kwargs)
2024-04-29T07:42:51.4715231Z   File "<boltons.funcutils.FunctionBuilder-4>", line 2, in build
2024-04-29T07:42:51.4715658Z   File "/opt/mambaforge/envs/bioconda/lib/python3.9/site-packages/bioconda_utils/cli.py", line 66, in wrapper
2024-04-29T07:42:51.4715898Z     func(*args, **kwargs)
2024-04-29T07:42:51.4716280Z   File "/opt/mambaforge/envs/bioconda/lib/python3.9/site-packages/bioconda_utils/cli.py", line 494, in build
2024-04-29T07:42:51.4716957Z     success = build_recipes(recipe_folder, config, recipes,
2024-04-29T07:42:51.4717400Z   File "/opt/mambaforge/envs/bioconda/lib/python3.9/site-packages/bioconda_utils/build.py", line 432, in build_recipes
2024-04-29T07:42:51.4721464Z     res = build(recipe=recipe,
2024-04-29T07:42:51.4721986Z   File "/opt/mambaforge/envs/bioconda/lib/python3.9/site-packages/bioconda_utils/build.py", line 185, in build
2024-04-29T07:42:51.4722358Z     pkg_test.test_package(pkg_path, base_image=base_image,
2024-04-29T07:42:51.4722827Z   File "/opt/mambaforge/envs/bioconda/lib/python3.9/site-packages/bioconda_utils/pkg_test.py", line 180, in test_package
2024-04-29T07:42:51.4723194Z     p = utils.run(cmd, env=env, cwd=d, mask=False, live=live_logs)
2024-04-29T07:42:51.4723546Z   File "/opt/mambaforge/envs/bioconda/lib/python3.9/tempfile.py", line 839, in __exit__
2024-04-29T07:42:51.4735194Z     self.cleanup()
2024-04-29T07:42:51.4736593Z   File "/opt/mambaforge/envs/bioconda/lib/python3.9/tempfile.py", line 843, in cleanup
2024-04-29T07:42:51.4737415Z     self._rmtree(self.name)
2024-04-29T07:42:51.4737661Z   File "/opt/mambaforge/envs/bioconda/lib/python3.9/tempfile.py", line 825, in _rmtree
2024-04-29T07:42:51.4737893Z     _shutil.rmtree(name, onerror=onerror)
2024-04-29T07:42:51.4738706Z   File "/opt/mambaforge/envs/bioconda/lib/python3.9/shutil.py", line 734, in rmtree
2024-04-29T07:42:51.4738951Z     _rmtree_safe_fd(fd, path, onerror)
2024-04-29T07:42:51.4739201Z   File "/opt/mambaforge/envs/bioconda/lib/python3.9/shutil.py", line 667, in _rmtree_safe_fd
2024-04-29T07:42:51.4739433Z     _rmtree_safe_fd(dirfd, fullname, onerror)
2024-04-29T07:42:51.4740199Z   File "/opt/mambaforge/envs/bioconda/lib/python3.9/shutil.py", line 667, in _rmtree_safe_fd
2024-04-29T07:42:51.4742409Z     _rmtree_safe_fd(dirfd, fullname, onerror)
2024-04-29T07:42:51.4742907Z   File "/opt/mambaforge/envs/bioconda/lib/python3.9/shutil.py", line 667, in _rmtree_safe_fd
2024-04-29T07:42:51.4743139Z     _rmtree_safe_fd(dirfd, fullname, onerror)
2024-04-29T07:42:51.4743342Z   [Previous line repeated 1 more time]
2024-04-29T07:42:51.4743568Z   File "/opt/mambaforge/envs/bioconda/lib/python3.9/shutil.py", line 690, in _rmtree_safe_fd
2024-04-29T07:42:51.4743825Z     onerror(os.unlink, fullname, sys.exc_info())
2024-04-29T07:42:51.4744055Z   File "/opt/mambaforge/envs/bioconda/lib/python3.9/tempfile.py", line 810, in onerror
2024-04-29T07:42:51.4744290Z     _resetperms(_os.path.dirname(path))
2024-04-29T07:42:51.4744519Z   File "/opt/mambaforge/envs/bioconda/lib/python3.9/tempfile.py", line 285, in _resetperms
2024-04-29T07:42:51.4744768Z     _dont_follow_symlinks(_os.chmod, path, 0o700)
2024-04-29T07:42:51.4745014Z   File "/opt/mambaforge/envs/bioconda/lib/python3.9/tempfile.py", line 276, in _dont_follow_symlinks
2024-04-29T07:42:51.4745509Z     func(path, *args)
2024-04-29T07:42:51.4746139Z PermissionError: [Errno 1] Operation not permitted: '/tmp/tmpelr6wsq7/build/dist/x86_64-conda_cos6-linux-gnu/bin'
2024-04-29T07:45:48.4298658Z 
2024-04-29T07:45:48.4763251Z ##[error]Bash exited with code '1'.
2024-04-29T07:45:48.5563705Z ##[section]Finishing: Test
```

**Comment by martin-g** [comment]
The mulled-build issue is related to https://github.com/bioconda/bioconda-recipes/pull/46177#issuecomment-2079659198
But since the build is for Linux x86_64 it does not break with `Permission denied` but falls back to `docker pull involucro/tool:latest` Docker image and there is not enough disk space for it ...

**Comment by martin-g** [comment]
@TobyBaril Please try again!
The issue with the extra Docker container is resolved!

**Comment by martin-g** [comment]
```
DEBU Packing failed due to open build/dist/conda-meta/recon-1.08-h031d066_6/licenses/LICENSE: permission denied
```

This is similar to https://github.com/conda-forge/font-ttf-ubuntu-feedstock/issues/2 but for bioconda/recon

**Comment by martin-g** [comment]
https://github.com/bioconda/bioconda-recipes/pull/47602

**Comment by martin-g** [comment]
https://github.com/bioconda/bioconda-recipes/pull/47602#issuecomment-2088277251 - The `LICENSE` file now has read permissions for everyone !
Once https://github.com/bioconda/bioconda-recipes/pull/47602 is merged you could try again !

**Comment by martin-g** [comment]
All green!

**Review by mencian — APPROVED** [review]

**Pull request merged**